src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,157 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+
+        if len(self.__assessment_packages_removed) == 0 and len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("No packages truncated")
+
+    def __get_status_payload_with_truncated_patches(self, status_file_payload_json_dumps):
+        """ Get truncated status file payload when status file byte size is more than 126kb """
+        status_file_size_in_bytes = self.__calc_status_size_on_disk(status_file_payload_json_dumps)  # calc complete_status_file_payload_json byte size on disk
+
+        if status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:  # perform truncation complete_status_file byte size > 126kb
+            self.composite_logger.log_debug("Status file limits evaluated. [StatusFileByteSize={0}] [InternalFileByteSize={1}]".format(
+                str(status_file_size_in_bytes), str(Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES)))
+            truncated_status_file = self.__create_truncated_status_file(status_file_size_in_bytes, status_file_payload_json_dumps)
+            status_file_payload_json_dumps = json.dumps(truncated_status_file)
+        return status_file_payload_json_dumps
+
+    def __create_truncated_status_file(self, status_file_size_in_bytes, complete_status_file_payload_json):
+        """ Truncate substatus message patch list when complete status file size is greater than 126kb """
+        self.composite_logger.log_verbose("Begin package list truncation")
+        truncated_status_file = json.loads(complete_status_file_payload_json)  # reload payload into python object
+        low_pri_index = None
+        assessment_substatus_index = self.__get_substatus_index(Constants.PATCH_ASSESSMENT_SUMMARY, truncated_status_file['status']['substatus'])
+        installation_substatus_index = self.__get_substatus_index(Constants.PATCH_INSTALLATION_SUMMARY, truncated_status_file['status']['substatus'])
+
+        if assessment_substatus_index is not None:      # If assessment data exists
+            self.__assessment_substatus_msg_copy = self.__get_substatus_message(truncated_status_file, assessment_substatus_index)
+            self.__assessment_packages_copy = self.__assessment_substatus_msg_copy['patches']
+
+        if installation_substatus_index is not None:    # If installation data exists
+            self.__installation_substatus_msg_copy = self.__get_substatus_message(truncated_status_file, installation_substatus_index)
+            self.__installation_packages_copy = self.__installation_substatus_msg_copy['patches']
+            low_pri_index = self.__get_installation_low_pri_index(self.__installation_packages_copy)
+
+        status_file_without_package_list_size = self.__size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # Deepcopy, fully copies the object to avoid reference modifications
```

Comment:
status_file_size_without_package_list

Reply:
I see **size_of_**max_packages_allowed_in_status used on line 885.
whereas status_file_without_package_list_**size** used here

Follow a consistent naming pattern all throughout your code, and in sync with the code base. In LPE, we have used ..._size everywhere else.

Rename size_of_max_packages_allowed_in_status on line 885 to max_package_size_allowed_in_status.

Another point of inconsistency: I see packages and patches being used interchangeably in code here. Please use patch everywhere

Reply:
I use patches instead of packages list, this reduce # of chars. done